src/Disks/DiskType.h

```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include <common/types.h>
+
+namespace DB
+{
+
+struct DiskType
```

Comment:
This looks like antipattern.
And there is zero comments in this file :sob: 

What does it mean "disk type" and why do we need to discriminate them?
Maybe remove this file and all its usages?